qendpoint-core/src/main/java/com/the_qa_company/qendpoint/core/storage/QEPDataset.java

```diff
@@ -26,7 +28,10 @@
  * @author Antoine Willerval
  * @see QEPCore
  */
-public class QEPDataset implements Closeable {
+public class QEPDataset implements Closeable, Serializable {
+
+	@Serial
+	private static final long serialVersionUID = 7525689572432598258L;
```

Comment:
There are some requirements for serializable classes in RDF4J that were not explicitly tested for. From what I remember when migrating the code to RDF4J 5.0.0 this class is somehow used transitively from within a query plan.